src/main/java/org/folio/rest/core/RestClient.java

```diff
@@ -27,13 +32,66 @@
 public class RestClient {
   private static final Logger LOGGER = LogManager.getLogger(RestClient.class);
   private static final String ERROR_MESSAGE = "errorMessage";
+  private static final String SEARCH_ENDPOINT = "%s?limit=%s&offset=%s%s";
+  private static final String EXCEPTION_CALLING_ENDPOINT_MSG = "Exception calling {} {} {}";
 
   private final String baseEndpoint;
 
   public RestClient(String baseEndpoint) {
     this.baseEndpoint = baseEndpoint;
   }
 
+  public <T> Future<T> get(String query, int offset, int limit, RequestContext requestContext, Class<T> responseType) {
+    String endpoint = String.format(SEARCH_ENDPOINT, baseEndpoint, limit, offset, buildQueryParam(query, LOGGER));
+    return get(requestContext, endpoint, responseType);
+  }
+
+  public <T> Future<T> getById(String id, RequestContext requestContext, Class<T> responseType) {
+    String endpoint = baseEndpoint + "/" + id;
```

Comment:
nitpick: you can use numerous utilities to ensure a well formed URL. Example: https://www.baeldung.com/java-url